torchtune/modules/loss/chunked_cross_entropy.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+from typing import List
+
+import torch
+
+
+class ChunkedCrossEntropyLoss(torch.nn.Module):
+    """
+    TODO: explain why it saves memory
+    """
+
+    def __init__(self, num_output_chunks: int = 16, ignore_index: int = -100):
+        super(ChunkedCrossEntropyLoss, self).__init__()
+        self.num_output_chunks = num_output_chunks
+        self.ignore_index = ignore_index
+        self.cross_entropy_loss = torch.nn.CrossEntropyLoss(
+            reduction="sum", ignore_index=self.ignore_index
```

Comment:
Why the change in reduction? I think previously we were using the default?

Reply:
The default is mean. I can do the mean, sum the means, and divide by the number of chunks. It should be equivalent and probably a bit faster

Reply:
I think your current way is more correct. When there are different paddings across samples in a batch (i.e. number of ignore_index is different), taking the mean of (mean of each chunk) != (total loss of non_padded tokens / no. of non_padded_tokens)

Reply:
oh, right! I would have to do a few more operations for it to be numerically the same. Thats why i computed the sum instead of the mena.